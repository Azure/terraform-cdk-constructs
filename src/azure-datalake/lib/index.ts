export * from "./filesystem";
export * from "./path";
export * from "./datalake";
