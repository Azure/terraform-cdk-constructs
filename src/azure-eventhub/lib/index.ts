export * from "./authorization";
export * from "./eventhubcluster";
export * from "./consumer";
export * from "./instance";
export * from "./kusto-connection";
export * from "./namespace";
