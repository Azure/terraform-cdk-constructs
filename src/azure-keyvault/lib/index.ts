export * from './certificate'
export * from './key'
export * from './policy'
export * from './secret'
export * from './vault'

