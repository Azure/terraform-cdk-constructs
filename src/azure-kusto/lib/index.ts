export * from "./cluster";
export * from "./compute-specification";
export * from "./database";
