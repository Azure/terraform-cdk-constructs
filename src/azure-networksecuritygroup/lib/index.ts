export * from './preconfigured-rules';
export * from './securitygroup';
