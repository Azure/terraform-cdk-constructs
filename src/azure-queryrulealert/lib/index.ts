export * from './query-rule-alert'