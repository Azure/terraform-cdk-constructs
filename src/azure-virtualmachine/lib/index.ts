export * from './image-references';
export * from './vm';
