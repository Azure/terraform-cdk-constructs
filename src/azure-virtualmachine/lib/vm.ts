import {
  LinuxVirtualMachine,
  LinuxVirtualMachineSourceImageReference,
  LinuxVirtualMachineOsDisk,
  LinuxVirtualMachineAdminSshKey,
  LinuxVirtualMachineIdentity,
  LinuxVirtualMachineSecret,
  LinuxVirtualMachineAdditionalCapabilities,
} from "@cdktf/provider-azurerm/lib/linux-virtual-machine";
import { NetworkInterface } from "@cdktf/provider-azurerm/lib/network-interface";
import { PublicIp } from "@cdktf/provider-azurerm/lib/public-ip";
import { ResourceGroup } from "@cdktf/provider-azurerm/lib/resource-group";
import { Subnet } from "@cdktf/provider-azurerm/lib/subnet";
import { VirtualMachineExtension } from "@cdktf/provider-azurerm/lib/virtual-machine-extension";
import {
  WindowsVirtualMachine,
  WindowsVirtualMachineOsDisk,
  WindowsVirtualMachineSourceImageReference,
} from "@cdktf/provider-azurerm/lib/windows-virtual-machine";
import * as cdktf from "cdktf";
import { Construct } from "constructs";

import { WindowsImageReferences } from "./image-references";
import { Network } from "../../azure-virtualnetwork/lib/network";
import { AzureResource } from "../../core-azure/lib";

export interface WindowsVMProps {
  /**
   * The Azure location where the virtual machine should be created.
   * @default "eastus"
   */
  readonly location?: string;

  /**
   * The name of the virtual machine.
   * @default - Uses the name derived from the construct path.
   */
  readonly name?: string;

  /**
   * The name of the resource group in which the virtual machine will be created.
   */
  readonly resourceGroup: ResourceGroup;

  /**
   * The size of the virtual machine.
   * @default "Standard_B2s"
   */
  readonly size?: string;

  /**
   * The admin username for the virtual machine.
   */
  readonly adminUsername: string;

  /**
   * The admin password for the virtual machine.
   */
  readonly adminPassword: string;

  /**
   * The source image reference for the virtual machine.
   * @default - Uses WindowsServer2022DatacenterCore.
   */
  readonly sourceImageReference?: WindowsVirtualMachineSourceImageReference;

  /**
   * The ID of the source image for the virtual machine.
   */
  readonly sourceImageId?: string;

  /**
   * Tags to apply to the virtual machine.
   */
  readonly tags?: { [key: string]: string };

  /**
   * The OS disk configuration for the virtual machine.
   * @default - Uses a disk with caching set to "ReadWrite" and storage account type "Standard_LRS".
   */
  readonly osDisk?: WindowsVirtualMachineOsDisk;

  /**
   * The subnet in which the virtual machine will be placed.
   * @default - Uses the default subnet from a new virtual network.
   */
  readonly subnet?: Subnet;

  /**
   * The allocation method for the public IP.
   */
  readonly publicIPAllocationMethod?: string;

  /**
   * Custom data to pass to the virtual machine upon creation.
   */
  readonly customData?: string;

  /**
   * Custom data to bootstrap the virtual machine. Automatically triggers Azure Custom Script extension to deploy code in custom data.
   */
  readonly boostrapCustomData?: string;

  /**
   * Bootdiagnostics settings for the VM.
   */
  readonly bootDiagnosticsStorageURI?: string;
}

export class WindowsVM extends AzureResource {
  public readonly props: WindowsVMProps;
  public resourceGroup: ResourceGroup;
  public id: string;
  public readonly name: string;
  public readonly publicIp?: string;

  /**
   * Represents a Windows-based Virtual Machine (VM) within Microsoft Azure.
   *
   * This class is designed to provision and manage a Windows VM in Azure, allowing for detailed configuration including
   * the VM's size, the operating system image, network settings, and administrative credentials. It supports customization
   * of the OS disk, networking setup, and optional features like custom data scripts and boot diagnostics.
   *
   * @param scope - The scope in which to define this construct, typically representing the Cloud Development Kit (CDK) application.
   * @param id - The unique identifier for this instance of the Windows VM, used within the scope for reference.
   * @param props - Configuration properties for the Windows Virtual Machine, derived from the WindowsVMProps interface. These include:
   *                - `location`: The geographic location where the VM will be hosted (e.g., "eastus").
   *                - `name`: The name of the VM, which must be unique within the resource group.
   *                - `resourceGroup`: The ResourceGroup within which the VM will be created.
   *                - `size`: The size specification of the VM (e.g., "Standard_B2s").
   *                - `adminUsername`: The administrator username for accessing the VM.
   *                - `adminPassword`: The administrator password for accessing the VM.
   *                - `sourceImageReference`: A reference to the specific Windows image to be used for the VM.
   *                - `sourceImageId`: The identifier for a custom image to use for the VM.
   *                - `tags`: Key-value pairs for resource tagging.
   *                - `osDisk`: Configuration for the VM's operating system disk.
   *                - `subnet`: Specifies the subnet within which the VM will be placed.
   *                - `publicIPAllocationMethod`: The method used to allocate a public IP address to the VM.
   *                - `customData`: Scripts or commands passed to the VM at startup.
   *                - `bootstrapCustomData`: Custom data used to trigger the Azure Custom Script Extension for VM setup tasks.
   *                - `bootDiagnosticsStorageURI`: URI for storage where VM boot diagnostics are collected.
   *
   * Example usage:
   * ```typescript
   * const vm = new WindowsVM(this, 'MyWindowsVM', {
   *   resourceGroup: myResourceGroup,
   *   name: 'myVM',
   *   size: 'Standard_DS1_v2',
   *   adminUsername: 'adminuser',
   *   adminPassword: 'securepassword123',
   *   sourceImageReference: { publisher: 'MicrosoftWindowsServer', offer: 'WindowsServer', sku: '2019-Datacenter', version: 'latest' },
   *   osDisk: { caching: 'ReadWrite', storageAccountType: 'Standard_LRS' },
   *   subnet: mySubnet,
   *   tags: { environment: 'production' }
   * });
   * ```
   * This class initializes a Windows VM with the specified configurations, handling details like network setup, OS installation,
   * and security settings, thus providing a robust infrastructure for hosting applications on Windows Server environments.
   */
  constructor(scope: Construct, id: string, props: WindowsVMProps) {
    super(scope, id);

    this.props = props;
    this.resourceGroup = props.resourceGroup;

    // Default configurations for the virtual machine.
    const defaults = {
      name: props.name || this.node.path.split("/")[0],
      location: props.location || "eastus",
      size: props.size || "Standard_B2s",
      osDisk: props.osDisk || {
        caching: "ReadWrite",
        storageAccountType: "Standard_LRS",
      },
      sourceImageReference:
        props.sourceImageReference ||
        WindowsImageReferences.windowsServer2022DatacenterCore,
      subnet:
        props.subnet ||
        new Network(this, "vnet", {
          resourceGroup: props.resourceGroup,
        }).subnets.default,
    };

    // Create Public IP if specified.
    let publicIpId: string | undefined;
    if (props.publicIPAllocationMethod) {
      const azurermPublicIp = new PublicIp(this, "public-ip", {
        name: `pip-${defaults.name}`,
        resourceGroupName: props.resourceGroup.name,
        location: defaults.location,
        allocationMethod: props.publicIPAllocationMethod,
        tags: props.tags,
      });

      publicIpId = azurermPublicIp.id;
      this.publicIp = azurermPublicIp.ipAddress;
    }

    // Create the Network Interface.
    const azurermNetworkInterface = new NetworkInterface(this, "nic", {
      ...defaults,
      name: `nic-${defaults.name}`,
      resourceGroupName: props.resourceGroup.name,
      ipConfiguration: [
        {
          name: "internal",
          subnetId: defaults.subnet.id,
          privateIpAddressAllocation: "Dynamic",
          publicIpAddressId: publicIpId,
        },
      ],
      tags: props.tags,
    });

    // Base64 encode custom data if provided.
    const customData = props.customData || props.boostrapCustomData;
    const base64CustomData = customData
      ? Buffer.from(customData).toString("base64")
      : undefined;

    // Create the Windows Virtual Machine.
    const azurermWindowsVirtualMachine = new WindowsVirtualMachine(this, "vm", {
      ...defaults,
      resourceGroupName: props.resourceGroup.name,
      adminUsername: props.adminUsername,
      adminPassword: props.adminPassword,
      tags: props.tags,
      networkInterfaceIds: [azurermNetworkInterface.id],
      sourceImageId: props.sourceImageId,
      customData: base64CustomData,
      bootDiagnostics: { storageAccountUri: props.bootDiagnosticsStorageURI },
    });

    this.id = azurermWindowsVirtualMachine.id;
    this.name = azurermWindowsVirtualMachine.name;

    // Bootstrap VM with custom script extension if bootstrap custom data is provided.
    if (props.boostrapCustomData) {
      new VirtualMachineExtension(this, "script-ext", {
        name: `${this.name}-script-ext`,
        virtualMachineId: this.id,
        publisher: "Microsoft.Compute",
        type: "CustomScriptExtension",
        typeHandlerVersion: "1.10",
        protectedSettings:
          '{"commandToExecute": "rename  C:\\\\AzureData\\\\CustomData.bin  postdeploy.ps1 & powershell -ExecutionPolicy Unrestricted -File C:\\\\AzureData\\\\postdeploy.ps1"}',
      });
    }
  }
}

export interface LinuxVMProps {
  /**
   * The Azure location where the virtual machine should be created.
   * @default "eastus"
   */
  readonly location?: string;

  /**
   * The name of the virtual machine.
   * @default - Uses the name derived from the construct path.
   */
  readonly name?: string;

  /**
   * The name of the resource group in which the virtual machine will be created.
   */
  readonly resourceGroup: ResourceGroup;

  /**
   * The size of the virtual machine.
   * @default "Standard_B2s"
   */
  readonly size?: string;

  /**
   * The ID of the availability set in which the VM should be placed.
   */
  readonly availabilitySetId?: string;

  /**
   * Custom data to pass to the virtual machine upon creation.
   */
  readonly userData?: string;

  /**
   * An array of SSH keys for the admin user.
   */
  readonly adminSshKey?: LinuxVirtualMachineAdminSshKey[] | cdktf.IResolvable;

  /**
   * The availability zone in which the VM should be placed.
   */
  readonly zone?: string;

  /**
   * Managed identity settings for the VM.
   */
  readonly identity?: LinuxVirtualMachineIdentity;

  /**
   * Additional capabilities like Ultra Disk compatibility.
   */
  readonly additionalCapabilities?: LinuxVirtualMachineAdditionalCapabilities;

  /**
   * An array of secrets to be passed to the VM.
   */
  readonly secret?: LinuxVirtualMachineSecret[];

  /**
   * The admin username for the virtual machine.
   */
  readonly adminUsername?: string;

  /**
   * The admin password for the virtual machine.
   */
  readonly adminPassword?: string;

  /**
   * The source image reference for the virtual machine.
   * @default - Uses WindowsServer2022DatacenterCore.
   */
  readonly sourceImageReference?: LinuxVirtualMachineSourceImageReference;

  /**
   * The ID of the source image for the virtual machine.
   */
  readonly sourceImageId?: string;

  /**
   * Tags to apply to the virtual machine.
   */
  readonly tags?: { [key: string]: string };

  /**
   * The OS disk configuration for the virtual machine.
   * @default - Uses a disk with caching set to "ReadWrite" and storage account type "Standard_LRS".
   */
  readonly osDisk?: LinuxVirtualMachineOsDisk;

  /**
   * The subnet in which the virtual machine will be placed.
   * @default - Uses the default subnet from a new virtual network.
   */
  readonly subnet?: Subnet;

  /**
   * The allocation method for the public IP.
   */
  readonly publicIPAllocationMethod?: string;

  /**
   * Custom data to pass to the virtual machine upon creation.
   */
  readonly customData?: string;

  /**
   * Enable SSH Azure AD Login, required managed identity to be set.
   */
  readonly enableSshAzureADLogin?: boolean;

  /**
   * Bootdiagnostics settings for the VM.
   */
  readonly bootDiagnosticsStorageURI?: string;
}

export class LinuxVM extends AzureResource {
  // Properties of the AzureLinuxVirtualMachine class
  readonly props: LinuxVMProps;
  public resourceGroup: ResourceGroup;
  public id: string;
  public readonly name: string;
  public readonly publicIp?: string;

  /**
   * Represents a Linux-based Virtual Machine (VM) within Microsoft Azure.
   *
   * This class is designed to provision and manage a Linux VM in Azure, facilitating detailed configuration including
   * VM size, the operating system image, network settings, and administrative credentials. It supports custom data scripts,
   * SSH configurations, and optional features like managed identity and boot diagnostics.
   *
   * @param scope - The scope in which to define this construct, typically representing the Cloud Development Kit (CDK) application.
   * @param id - The unique identifier for this instance of the Linux VM, used within the scope for reference.
   * @param props - Configuration properties for the Linux Virtual Machine, derived from the LinuxVMProps interface. These include:
   *                - `location`: The geographic location where the VM will be hosted (e.g., "eastus").
   *                - `name`: The name of the VM, which must be unique within the resource group.
   *                - `resourceGroup`: The ResourceGroup within which the VM will be created.
   *                - `size`: The size specification of the VM (e.g., "Standard_B2s").
   *                - `availabilitySetId`: The ID of the availability set in which to include the VM.
   *                - `userData`: Custom data scripts to pass to the VM upon creation.
   *                - `adminSshKey`: SSH keys for secure access to the VM.
   *                - `zone`: The availability zone for deploying the VM.
   *                - `identity`: Managed identity settings for accessing other Azure services.
   *                - `additionalCapabilities`: Special capabilities like Ultra Disk support.
   *                - `sourceImageReference`: A reference to the specific Linux image to be used for the VM.
   *                - `sourceImageId`: The identifier for a custom image to use for the VM.
   *                - `tags`: Key-value pairs for resource tagging.
   *                - `osDisk`: Configuration for the VM's operating system disk.
   *                - `subnet`: Specifies the subnet within which the VM will be placed.
   *                - `publicIPAllocationMethod`: Method used to allocate a public IP address.
   *                - `customData`: Additional scripts or commands passed to the VM at startup.
   *                - `enableSshAzureADLogin`: Option to enable Azure AD login for SSH.
   *                - `bootDiagnosticsStorageURI`: URI for storage where VM boot diagnostics are collected.
   *
   * Example usage:
   * ```typescript
   * const linuxVM = new LinuxVM(this, 'MyLinuxVM', {
   *   resourceGroup: myResourceGroup,
   *   name: 'myVM',
   *   size: 'Standard_DS1_v2',
   *   adminUsername: 'adminuser',
   *   adminSshKey: [{ publicKey: 'ssh-rsa AAAAB...' }],
   *   sourceImageReference: { publisher: 'Canonical', offer: 'UbuntuServer', sku: '18.04-LTS', version: 'latest' },
   *   osDisk: { caching: 'ReadWrite', storageAccountType: 'Standard_LRS' },
   *   subnet: mySubnet,
   *   tags: { environment: 'production' }
   * });
   * ```
   * This class initializes a Linux VM with the specified configurations, handling details like network setup, OS installation,
   * and security settings, thus providing a robust infrastructure for hosting applications on Linux environments.
   */
  constructor(scope: Construct, id: string, props: LinuxVMProps) {
    super(scope, id);

    // Assigning the properties
    this.props = props;
    this.resourceGroup = props.resourceGroup;

    // Extracting the name from the node path
    const pathName = this.node.path.split("/")[0];

    // Setting default configurations for the virtual machine
    const defaults = {
      name: props.name || pathName,
      adminUsername: props.adminUsername || `admin${pathName}`,
      location: props.location || "eastus",
      size: props.size || "Standard_B2s",
      osDisk: props.osDisk || {
        caching: "ReadWrite",
        storageAccountType: "Standard_LRS",
      },
      sourceImageReference:
        props.sourceImageReference ||
        WindowsImageReferences.windowsServer2022DatacenterCore,
      subnet:
        props.subnet ||
        new Network(this, "vnet", {
          resourceGroup: props.resourceGroup,
        }).subnets.default,
    };

    // Create Public IP if specified
    let publicIpId: string | undefined;
    if (props.publicIPAllocationMethod) {
      const azurermPublicIp = new PublicIp(this, "public-ip", {
        name: `pip-${defaults.name}`,
        resourceGroupName: props.resourceGroup.name,
        location: defaults.location,
        allocationMethod: props.publicIPAllocationMethod,
        tags: props.tags,
      });

      publicIpId = azurermPublicIp.id;
      this.publicIp = azurermPublicIp.ipAddress;
    }

    // Create the Network Interface
    const azurermNetworkInterface = new NetworkInterface(this, "nic", {
      ...defaults,
      name: `nic-${defaults.name}`,
      resourceGroupName: props.resourceGroup.name,
      ipConfiguration: [
        {
          name: "internal",
          subnetId: defaults.subnet.id,
          privateIpAddressAllocation: "Dynamic",
          publicIpAddressId: publicIpId,
        },
      ],
      tags: props.tags,
    });

    // Create the Linux Virtual Machine
    const azurermLinuxVirtualMachine = new LinuxVirtualMachine(this, "vm", {
      ...defaults,
      resourceGroupName: props.resourceGroup.name,
      adminPassword: props.adminPassword,
      tags: props.tags,
      networkInterfaceIds: [azurermNetworkInterface.id],
      sourceImageId: props.sourceImageId,
      customData: props.customData
        ? Buffer.from(props.customData).toString("base64")
        : undefined,
      userData: props.userData
        ? Buffer.from(props.userData).toString("base64")
        : undefined,
      availabilitySetId: props.availabilitySetId,
      adminSshKey: props.adminSshKey,
      bootDiagnostics: { storageAccountUri: props.bootDiagnosticsStorageURI },
      zone: props.zone,
      identity: props.identity,
      additionalCapabilities: props.additionalCapabilities,
      secret: props.secret,
      disablePasswordAuthentication: props.adminPassword ? false : true,
    });

    // Assigning the VM's ID and name to the class properties
    this.id = azurermLinuxVirtualMachine.id;
    this.name = azurermLinuxVirtualMachine.name;

    // Enable SSH Azure AD Login if specified
    if (props.enableSshAzureADLogin) {
      new VirtualMachineExtension(this, "AADSSHlogin", {
        name: "AADSSHLoginForLinux",
        virtualMachineId: this.id,
        publisher: "Microsoft.Azure.ActiveDirectory",
        type: "AADSSHLoginForLinux",
        typeHandlerVersion: "1.0",
        tags: props.tags,
      });
    }
  }
}
