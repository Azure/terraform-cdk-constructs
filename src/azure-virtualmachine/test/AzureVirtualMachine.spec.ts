import { Testing, TerraformStack} from 'cdktf';
import 'cdktf/lib/testing/adapters/jest';
import {AzureVirtualMachine} from '..';
import {AzurermProvider} from "@cdktf/provider-azurerm/lib/provider";

describe('Azure Virtual Machine With Defaults', () => {
  let stack: TerraformStack;
  let fullSynthResult: any;

  beforeEach(() => {
    const app = Testing.app();
    stack = new TerraformStack(app, "testAzureVMWithDefaults");

    new AzurermProvider(stack, "azureFeature", {features: {}});

    new AzureVirtualMachine(stack, "testVirtualMachine", {
      resource_group_name: "testResourceGroup",
      admin_username: "testAdmin",
      admin_password: "testPassword&@34$$123",
    });

    fullSynthResult = Testing.fullSynth(stack); // Save the result for reuse
  });

  it("renders an Azure Network Security Group with defaults and checks snapshot", () => {
    expect(
      Testing.synth(stack)
    ).toMatchSnapshot(); // Compare the already prepared stack
  });

  it("check if the produced terraform configuration is valid", () => {
    expect(fullSynthResult).toBeValidTerraform(); // Use the saved result
  });

  it("check if this can be planned", () => {
    expect(fullSynthResult).toPlanSuccessfully(); // Use the saved result
  });
});



// describe('Azure Network Security Group Example', () => {
  
//   it("renders the Example Network Security Group and checks snapshot", () => {
  
//     expect(
//       Testing.synth(new exampleAzureNetworkSecurityGroup(Testing.app(), "testNetworkSecurityGroupExample"))
      
//     ).toMatchSnapshot();
// });

//   it("check if the produced terraform configuration is valid", () => {
//     // We need to do a full synth to plan the terraform configuration
//     expect(Testing.fullSynth(new exampleAzureNetworkSecurityGroup(Testing.app(), "testNetworkSecurityGroupExample"))).toBeValidTerraform();
//   });

//   it("check if this can be planned", () => {
  
    
//     // We need to do a full synth to plan the terraform configuration
//     expect(Testing.fullSynth(new exampleAzureNetworkSecurityGroup(Testing.app(), "testNetworkSecurityGroupExample"))).toPlanSuccessfully();
//   });
// });