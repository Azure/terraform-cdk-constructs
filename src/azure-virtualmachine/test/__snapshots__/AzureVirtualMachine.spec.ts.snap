// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Azure Virtual Machine With Defaults renders an Azure Network Security Group with defaults and checks snapshot 1`] = `
"{
  "output": {
    "vm_id": {
      "value": "\${azurerm_windows_virtual_machine.testVirtualMachine_vm_3FD746F5.id}"
    }
  },
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_network_interface": {
      "testVirtualMachine_nic_45477852": {
        "ip_configuration": [
          {
            "name": "internal",
            "private_ip_address_allocation": "Dynamic",
            "subnet_id": "\${azurerm_subnet.testVirtualMachine_vnet_default_4A281376.id}"
          }
        ],
        "location": "eastus",
        "name": "nic-testAzureVMWithDefaults",
        "resource_group_name": "testResourceGroup"
      }
    },
    "azurerm_subnet": {
      "testVirtualMachine_vnet_default_4A281376": {
        "address_prefixes": [
          "10.1.0.0/24"
        ],
        "name": "default",
        "resource_group_name": "testResourceGroup",
        "virtual_network_name": "\${azurerm_virtual_network.testVirtualMachine_vnet_vnet-testAzureVMWithDefaults_B2341822.name}"
      }
    },
    "azurerm_virtual_network": {
      "testVirtualMachine_vnet_vnet-testAzureVMWithDefaults_B2341822": {
        "address_space": [
          "10.0.0.0/16"
        ],
        "location": "eastus",
        "name": "vnet-testAzureVMWithDefaults",
        "resource_group_name": "testResourceGroup"
      }
    },
    "azurerm_windows_virtual_machine": {
      "testVirtualMachine_vm_3FD746F5": {
        "admin_password": "testPassword&@34$$123",
        "admin_username": "testAdmin",
        "location": "eastus",
        "name": "testAzureVMWithDefaults",
        "network_interface_ids": [
          "\${azurerm_network_interface.testVirtualMachine_nic_45477852.id}"
        ],
        "os_disk": {
          "caching": "ReadWrite",
          "storage_account_type": "Standard_LRS"
        },
        "resource_group_name": "testResourceGroup",
        "size": "Standard_B2s",
        "source_image_reference": {
          "offer": "WindowsServer",
          "publisher": "MicrosoftWindowsServer",
          "sku": "2022-Datacenter-Core",
          "version": "latest"
        }
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.66.0"
      }
    }
  }
}"
`;
